### PR TITLE
MatchHeader

### PR DESCRIPTION
**Pull request by blukai**
image

Region, skill and buttons will move to another location at the bottom (under overview table)

### CODE REVIEW

**Comment by mike-robertson** [comment]
Looks good! I'll try to review sometime today.

**Comment by howardchung** [comment]
@bguzryanto who originally implemented match header

**Comment by dartharaknus** [comment]
Can we add region, skill and matchID to the header too? I don't think they should be separated.

**Comment by blukai** [comment]
I do not consider it necessary to reflect this in the header since nothing prevents scroll down the page a bit. If it will be not so convenient, then we can display it in header as it always was. 🍀

File: src/components/Match/MatchHeader/MatchHeader.css [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  font-size: 28px;
+  padding-top: 10px;
+  -webkit-filter: drop-shadow(0 0 15px var(--colorYelor));
+  filter: drop-shadow(0 0 15px var(--colorYelor));
+}
+.score {
+  font-size: 48px;
+}
+
+.radiant div:first-of-type {
```
Comment (mike-robertson) [inline src/components/Match/MatchHeader/MatchHeader.css]: same thing for all these. We can use css-modules composes for shared styles.

```css
.team {
  & svg {
    width: 32px;
    height: 32px;
    margin: 0 15px;
    vertical-align: sub;
  }
}

.dire {
  composes: team;
  color: var(--colorDanger);
  & svg {
    fill: var(--colorDanger);
  }
  & div:first-of-type {
    float: left;
  }
}

.radiant {
  composes: team;
  color: var(--colorSuccess);
  & svg {
    fill: var(--colorSuccess);
  }
  & div:first-of-type {
    float: right;
  }
}
```

File: src/components/Match/MatchHeader/MatchHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import strings from 'lang';
+import Spinner from 'components/Spinner';
+import { IconRadiant, IconDire } from 'components/Icons';
+import styles from './MatchHeader.css';
+
+export default ({ match, loading }) => {
+  if (!loading) {
+    const killsRadint = [];
+    const killsDire = [];
+    let player;
```
Comment (mike-robertson) [inline src/components/Match/MatchHeader/MatchHeader.jsx]: You should declare this in the for loop if it is scoped to the for loop

File: src/components/Match/MatchHeader/MatchHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    for (let i = 0; i < match.players.length; i += 1) {
```
Comment (mike-robertson) [inline src/components/Match/MatchHeader/MatchHeader.jsx]: Actually, just do this functionally:
```js
const kills = match.players.reduce(
  (kills, player) => {
    if (isRadiant(player.player_slot) {
      kills.radiant.push(player.kills);
    } else {
      kills.dire.push(player.kills);
    }
    return kills;
  },
  {
    radiant: [],
    dire: [],
  }
);
```

File: src/components/Match/MatchHeader/MatchHeader.css [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  & svg {
+    width: 32px;
+    height: 32px;
+    margin: 0 15px;
+    vertical-align: sub;
+  }
+}
+
+.radiant {
+  composes: team;
```
Comment (mike-robertson) [inline src/components/Match/MatchHeader/MatchHeader.css]: Can we make this responsive? Right now, it sort of breaks after 760px, and it starts to look sort of bad at ~820ish

File: src/components/Match/Match.jsx [hunk 5]
```diff
@@ -48,13 +42,8 @@ class RequestLayer extends React.Component {
...
     return (
       <div>
-        <MatchHeader match={match} user={this.props.user} />
-        <div style={{ marginTop: 25 }}>
-          <TabBar
-            info={info}
-            tabs={matchPages(matchId)}
-          />
-        </div>
+        <MatchHeader match={match} loading={this.props.loading} />
```
Comment (howardchung) [inline src/components/Match/Match.jsx]: if we aren't going to pass user here, we can remove the selector?

File: src/components/Match/MatchHeader/MatchHeader.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import { Row, Col } from 'react-flexbox-grid';
+import { transformations, isRadiant } from 'utility';
+import strings from 'lang';
+import Spinner from 'components/Spinner';
+import { IconRadiant, IconDire } from 'components/Icons';
+import styles from './MatchHeader.css';
+
+export default ({ match, loading }) => {
+  if (!loading) {
+    const kills = match.players.reduce(
```
Comment (howardchung) [inline src/components/Match/MatchHeader/MatchHeader.jsx]: I think it's cleaner to just iterate twice here and use .map and .reduce to get your final values.  Sure, you're looping twice, but number of players is not scaling.

File: src/components/Match/MatchHeader/MatchHeader.jsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        return kills;
+      },
+      {
+        radiant: [],
+        dire: [],
+      }
+    );
+
+    return (
+      <header className={styles.header}>
```
Comment (howardchung) [inline src/components/Match/MatchHeader/MatchHeader.jsx]: Consider adding space to support team names (which is something I want to add at some point).  Perhaps just make some empty elements and test layout by putting in fake team names.

**Comment by blukai** [comment]
Normal view
image

1024px
image

**Pull request merged**